glib build fails (both for python2x and 3x)

### Steps to reproduce the issue

```console
$ spack install glib 
```
in the log we are seeing the following errors:

````
1831  /usr/bin/install -c -m 644 ./html/up-insensitive.png
1832  /usr/bin/install -c -m 644 ./html/up.png
1833 Traceback (most recent call last):
1834   File "/usr/bin/gtkdoc-rebase", line 28, in <module>
1835     from gtkdoc import common, config, rebase
1836   File "/usr/share/gtk-doc/python/gtkdoc/common.py", line 31, in <module>
1837     import six
1838 ImportError: No module named six
1839 Makefile:1034: recipe for target 'install-data-local' failed
1840 make[5]: *** [install-data-local] Error 1
1841 make[5]: Leaving directory '/tmp/sinan/spack-stage/spack-stage-9BmR48/glib-2.56.2/docs/reference/glib'
1842 Makefile:776: recipe for target 'install-am' failed
1843 make[4]: *** [install-am] Error 2
1844 make[4]: Leaving directory '/tmp/sinan/spack-stage/spack-stage-9BmR48/glib-2.56.2/docs/reference/glib'
1845 Makefile:484: recipe for target 'install-recursive' failed
1846 make[3]: *** [install-recursive] Error 1
1847 make[3]: Leaving directory '/tmp/sinan/spack-stage/spack-stage-9BmR48/glib-2.56.2/docs/reference'
1848 Makefile:485: recipe for target 'install-recursive' failed
1849 make[2]: *** [install-recursive] Error 1
1850 make[2]: Leaving directory '/tmp/sinan/spack-stage/spack-stage-9BmR48/glib-2.56.2/docs'
1851 Makefile:1279: recipe for target 'install-recursive' failed
1852 make[1]: *** [install-recursive] Error 1
1853 make[1]: Leaving directory '/tmp/sinan/spack-stage/spack-stage-9BmR48/glib-2.56.2'
1854 Makefile:1748: recipe for target 'install' failed
````
[spack-build.out.txt](https://github.com/spack/spack/files/2563044/spack-build.out.txt)


### Platform and user environment

Please report your OS here:
```commandline
$ uname -a
Linux storm 4.15.0-36-generic #39-Ubuntu SMP Mon Sep 24 16:19:09 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -d
Description:    Linux Mint 19 Tara
``` 
This error should be also reproduceable on Ubuntu 18.04.1

I think this is not a major problem as error has to do with the building of the documentation. If I understand correctly, glib is attempting to use already existing gtk documentation in the system.

--- a/var/spack/repos/builtin/packages/glib/package.py
+++ b/var/spack/repos/builtin/packages/glib/package.py
@@ -72,6 +72,18 @@ class Glib(AutotoolsPackage):
                 os.path.basename(self.spec['python'].command.path))
             )
         args.extend(self.enable_or_disable('tracing'))
+        # glib should not use the globally installed gtk-doc. Otherwise,
+        # gtk-doc can fail with Python errors such as "ImportError: No module
+        # named site". This is due to the fact that Spack sets PYTHONHOME,
+        # which can confuse the global Python installation used by gtk-doc.
+        args.append('--disable-gtk-doc-html')
+        # glib uses gtk-doc even though it should be disabled if it can find
+        # its binaries. Override the checks to use the true binary.
+        true = which('true')
+        args.append('GTKDOC_CHECK={0}'.format(true))
+        args.append('GTKDOC_CHECK_PATH={0}'.format(true))
+        args.append('GTKDOC_MKPDF={0}'.format(true))
+        args.append('GTKDOC_REBASE={0}'.format(true))
         return args
 
     @property

